pallets/liquidity-provider/src/lib.rs

```diff
@@ -63,33 +76,54 @@ decl_module! {
             asset: Asset,
             balance: Balance<T>,
             offer_rates: Vec<OfferRateTy<T>>
-        ) -> dispatch::DispatchResult
+        ) -> DispatchResult
         {
             match asset {
-                Asset::Usdv => return Err(crate::Error::<T>::MustNotBeUsdv.into()),
+                Asset::Usdv => Err(crate::Error::<T>::MustNotBeUsdv.into()),
                 Asset::Collateral(collateral) => {
                     let who = ensure_signed(origin)?;
-                    update_account_rates::<T>(&who, asset, offer_rates);
-                    do_attest::<T>(who.clone(), Asset::Usdv, balance)?;
+                    Self::update_account_rates(&who, asset, offer_rates);
+                    Self::do_attest(who.clone(), Asset::Usdv, balance)?;
                     T::Collateral::deposit(collateral.into(), &who, balance)?;
                     T::Collateral::reserve(collateral.into(), &who, balance)?;
                     Self::deposit_event(RawEvent::Attestation(who, collateral.into()));
                     Ok(())
+                },
+                Asset::Btc | Asset::Cop | Asset::Ves => {
+                    todo!()
                 }
             }
         }
 
+        #[weight = 0]
+        pub fn submit_pair_prices(
```

Comment:
`submit_pair_prices` is called internally by the `offchain_worker` method entry-point. Even if `submit_pair_prices` were private, which I don't think it could because workers are external services,  everyone could simply call `offchain_worker` to the same end result.

> anyone can trigger a pair price change no?

Yes. Every node in the blockchain has access to this method. Not sure how to whitelist participants but it might be possible